launch.go

```diff
@@ -72,6 +73,32 @@ func init() {
 	client = retryablehttp.NewClient()
 }
 
+/* has HTTP or HTTPS protocol
+targetUrl => URL
+*/
+func hasHTTPProtocol(targetUrl *url.URL) bool {
+	return strings.Contains(targetUrl.Scheme, "http") || strings.Contains(targetUrl.Scheme, "https")
+}
+
+/* make pushgateway url
+baseUrl => base url for pushgateway
+buildID => sd build id
+*/
+func makePushgatewayUrl(baseUrl string, buildID int) (*url.URL, error) {
+	var pushgatewayUrl string = baseUrl
+	u, err := url.Parse(pushgatewayUrl)
+	if err != nil {
+		log.Printf("makePushgatewayUrl: failed to parse url [%v], buildId:[%v], error:[%v]", pushgatewayUrl, buildID, err)
+		return nil, err
+	}
+
+	if !hasHTTPProtocol(u) {
```

Comment:
Is it better to set http or https, if there is no Schema?

Reply:
I fixed it.